test/jellyfish_web/controllers/component_controller_test.exs

```diff
@@ -0,0 +1,72 @@
+defmodule JellyfishWeb.ComponentControllerTest do
+  use JellyfishWeb.ConnCase
+
+  setup %{conn: conn} do
+    room_conn = post(conn, Routes.room_path(conn, :create))
+    assert %{"id" => id} = json_response(room_conn, 201)["data"]
+    {:ok, %{conn: put_req_header(conn, "accept", "application/json"), room_id: id}}
+  end
+
+  describe "create component" do
+    test "renders component when data is valid", %{conn: conn, room_id: room_id} do
+      conn = post(conn, Routes.component_path(conn, :create, room_id), component_type: "hls")
+      assert %{"id" => id} = json_response(conn, 201)["data"]
+
+      conn = get(conn, Routes.room_path(conn, :show, room_id))
+
+      assert %{
+               "id" => ^room_id,
+               "components" => [
+                 %{"id" => ^id, "type" => "Elixir.Membrane.RTC.Engine.Endpoint.HLS"}
```

Comment:
Shouldn't the component type here be `"hls"`? I don't think we should return elixir module names as types.

Reply:
Nope because we want to get the type from the engine.